### DISCUSSION TITLE
All geners1 empty

### DISCUSSION THREAD

**Discussion by Taglia74** [jamulussoftware/jamulus#3269] [Technical Support & Help]
Goodmorning it is a couple of mounth that I see the server 1 list copletely empty (please see the attached picure)
I've istall jamulus again and again to try to solve the issue but nothing
I can see the lists in other geners pages but not in a page of all geners 1
May you help me?

![Screenshot 2024-04-21 110856](https://github.com/jamulussoftware/jamulus/assets/167677874/815ecfe1-35e2-4408-901f-18e9ecbed774)

**Comment by ann0see** [comment]
This is most likely a network issue. Can you ping anygenre1.jamulus.io

**Comment by Taglia74** [comment]
> This is most likely a network issue. Can you ping anygenre1.jamulus.io

Hello thanks about your feedback. How can I ping anygenre1.jamulus.io?
Best regards

**Comment by ann0see** [comment]
Hi,

This depends on the operating system you are using. Usually, you'd open terminal and type ping anygenre1.jamulus.io 

If it needs a long time doing nothing, gives timeouts or similar, there's a network (reachability) issue. If it gives reasonable times the connection is fine. 

See https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/ping for windows. 

If this fails, check your firewall settings locally. Maybe some antivirus blocks Jamulus.
If that isn't the case, the next step would be checking your router.

**Comment by Taglia74** [comment]
Hi 

thanks for your answare. I've tested the pin and the result is 32ms so I think that's alla right

I've not particular setting on router (I laft as it is from network provider)

The strange things is that till 2 or 3 mounth ago it work without problem, after that when I istalled a new jamulus version it stop to work only server 1

I've not antitivirus istalled and firewall offf but it doesn't work anyway.

Is it necessary to set a particular things on router?

Thanks a lot

**Comment by ann0see** [comment]
Ok. That means that the connection works - at least for ICMP packets (which are used for Ping). Usually you do not need to set up anything on the router - but maybe there's still something strange happening on your specific one.

Next step would be if you can connect to anygenre1.jamulus.io without a directory. To do that, enter anygenre1.jamulus.io into "Indirizzo del Server".
If you can connect to the server this could be some fragmentation issue (not sure if that causes an empty list). If you cannot connect this could hint at a more easy to debug problem. I assume you will be able to connect since Anygenre2 is hosted on the same machine as Anygenre1 (and the ping worked) but there may be some blocked ports. 

Moreover, maybe unclick "Visualizza Musicisti".

From the screenshots it seems as if you have Windows 11 as note for others. 

Could you please:
1. Tell us what router you are using
2. Tell us which exact Jamulus version and OS you are using
3. If you have other devices you could test Jamulus on.

**Comment by Taglia74** [comment]
I tested connection to apply into indirizzo  anygenre1.jamulus.io    and it work.

I've two pc, one with win 10 in LAN (Jamulus version 3.10.0) and one pc with win11 in wireless (Jamulus version 3.10.0)

Both the same problem

I've tried to unflag "visualizza musicisti" buth nothing change

My router:

**Comment by ann0see** [comment]
As this happens with multiple devices it points to an issue with either your router or ISP then. Could you please send the router name as I cannot see what you sent? Also firewall settings in it?

I'm afraid that we can't do much if the issue is on your ISPs side except for getting contact with them and telling them about connectivity problems. 

I have the feeling that it could also be related to UDP fragmentation. 

What you could do is to restart your router and let it off for ~ 5 minutes. Maybe then it handles out another IP and you get different routing.

**Comment by ann0see** [comment]
Also Wireless (if it's WiFi) will usually give you severely degraded performance.

**Comment by Taglia74** [comment]
Please here next the information that you answer.

Best regards


Sky Wifi Hub is the router name

Model:SR203

DSL Versione software:A2pvfbH045q.d27h

 

 

IPV6 router firewall setting output traffic:

block all unrelated traffic and enable IDS

 

 

IPV4 router firewall setting output traffic:

blocks the protocols below and enables IDS 

IDENT (porta 113)

**Comment by Taglia74** [comment]
Yes thanks, I'm using jamulus only in the lan connected pc

**Comment by ann0see** [comment]
Interesting. If IDS is an intrusion detection system then that may block the data. But you should not disable this. Maybe there's a setting to tweak it. Unfortunately I have no experience with this kind of routers. 

There's something more advanced to track down the issue: wireshark. You can check if your device gets back the responses of the server. 

You'd install wireshark from the bottom of https://www.wireshark.org/, and 
https://github.com/softins/jamulus-wireshark/ then you run wireshark and log all your network traffic. Open up Jamulus and the server list, now check if you see responses from the servers.

This video may be an introduction to wireshark: https://www.youtube.com/watch?v=lb1Dw0elw0Q and https://www.lifewire.com/wireshark-tutorial-4143298

I'm afraid that the real fix would be to talk to your ISP. But this above helps collecting data.

**Comment by ann0see** [comment]
Converted to a discussion as this is most likely not a bug. Also gives more visibility to the community.

**Reply by ann0see** [reply]
@Taglia74 In case you need a new link, the discussion is here: https://github.com/orgs/jamulussoftware/discussions/3269

**Comment by Taglia74** [comment]
Goodmorning, I've istalled wireshark but I can't notice nothing wrong (I'm not a computer engineer) 

I've put down to test also IDS in the router but nothing change.

Before to install last version of jamulus I didn't have this issue and I had the same network provider.

If you wanna do a remote session directly on my desk pc to see what happen to me no problem.

Best regards

Andrea

**Reply by ann0see** [reply]
Thanks for your answer.

Some questions:
How do you classify that nothing is wrong? What does wireshark give you back? Do you see responses from all servers? Can you post a screenshot or better a (privacy censored) log?

**Reply by ann0see** [reply]
> If you wanna do a remote session directly on my desk pc to see what happen to me no problem.

I'd only do that if really needed. 
I think @softins as network specialist would be better anyway (but they're on holiday)

**Comment by Taglia74** [comment]
Thanks

Last day when I start wireshark I can't see any red raw.

Today sometime I can see a red raw.

How can I extrapolate a privacy log file from wireshark?

thanks

**Reply by ann0see** [reply]
You can export normally and remove sensitive information by hand via a text editor. (E.g remove your external IP address)


https://www.wireshark.org/docs/wsug_html_chunked/ChIOExportSection.html

**Reply by ann0see** [reply]
Also the link I gave https://www.lifewire.com/wireshark-tutorial-4143298 should tell you exporting/logging data

**Reply by ann0see** [reply]
A red row may not show what we want to know. We'd want to know if packets from the servers come back correctly. If they don't arrive, we don't see them in wireshark.

**Comment by Taglia74** [comment]
How can I collect a wireshark log file about Jamulus activity?

**Comment by Taglia74** [comment]
Ahh ok

Is it that what you need?

**Reply by ann0see** [reply]
Yes. Please upload it to GitHub.

**Comment by Taglia74** [comment]
< DELETED >
Is it that ok?

**Reply by ann0see** [reply]
Ok. It loads fine in wireshark locally. I will check if I see something.

**Reply by ann0see** [reply]
Ok. This:

![grafik](https://github.com/jamulussoftware/jamulus/assets/20726856/fe76dcd5-ad6f-4719-ab7a-8661f1b959b2)

looks like the server list came back, but had invalid length - that's a fragmentation issue.

**Reply by ann0see** [reply]
Ok. From the log you provides it indeed seems that this could be a fragmentation issue as on CLM_REQ_SERVER_LIST we see a response from the server but with invalid length (and the content looks like a server list).

This does **not** have an easy fix. You could try to restart your router to find a new route which maybe doesn't fragment. Otherwise maybe there's an update you could revert on your router?

Otherwise, @softins is working on it (https://github.com/orgs/jamulussoftware/discussions/1058) and https://github.com/jamulussoftware/jamulus/issues/3242

What you can do in the meantime: Check the server list from https://explorer.jamulus.io and manually enter the IPs of the servers you want to join.

**Reply by ann0see** [reply]
I've deleted the file now as it may contain sensitive information like your PCs mac address. (It's not really personal information, but allows to see the vendor of your device etc). Also there are other packets (even though encrypted).

**Reply by softins** [reply]
I'm back from holiday now, although mainly busy with family stuff for the next few days.

@Taglia74 what make and model of router to do you have?

I have seen similar behaviour with my Draytek router when I had "hardware acceleration" enabled. It would attempt to re-assemble incoming fragmented IP packets before forwarding and re-fragmenting them. However, it would truncate the UDP payload at about 2000 bytes, although keeping the original length in the packet headers. The Wireshark trace would show the length mismatches in exactly the same way as the image above. I observed the same empty genres in Jamulus as a result.

When I disabled hardware acceleration, the incoming fragmented packets were correctly passed straight through the router, with the correct NAT mapping, allowing the receiving host endpoint to perform its own fragmented IP re-assembly correctly. Jamulus then showed all the directory lists correctly. So I have left acceleration disabled, and the router still performs with ample speed and no difficulties with fragmentation.

**Comment by Taglia74** [comment]
Thanks to all

Sky Wifi Hub is the router name

Model:SR203 DSL

Software version:A2pvfbH045q.d27h

I can't found out no one setting in the router about hardware acceleration....it is router that net provider give me in use....I didn't set it. I have access to the setting but I can't found any accelleration hardware....sorry

Best regards

**Comment by Taglia74** [comment]
Ok thanks a lot

**Comment by Taglia74** [comment]
That is a good work around hoping that will be fixed issue.

thanks

Andrea

**Comment by Wubbidubbi** [comment]
Hey.

I just want to let you know, that I have the same issue.
I will try the workaround. 

My Provider and Router are from Vodafone.

Thanks for your work.

**Reply by Wubbidubbi** [reply]
The workaround mentioned above worked for me.

https://explorer.jamulus.io/

Here you have the server lists. You can look up the ip address and then join directly.

**Comment by Ciccots** [comment]
I have the same issue on 4 PC, 2 are Mac and 2 are Win , connected to the same ISP (ILIAD).
Same problem with Android connected with wifi but if i switch off and i use 5G it works and i can see Anygenre1.
I've tried install previous versions of Jamulus on Win10 but nothing works , please help.
thanks

**Reply by softins** [reply]
This is not a bug in Jamulus, but rather a problem with the ISP or your router. For large UDP packets, such as one containing the list of servers in All Genres 1, Jamulus replies on the IP layer fragmenting and reassembling large packets. The fragmenting can happen at any point along the path, but the reassembly always happens at the receiving end. A few ISPs and a small proportion of networking equipment do not correctly pass fragmented IP packets, and this is what can result in large genre lists appearing empty.

It would appear that either ILIAD is to blame, or your router that is connected to it. That is confirmed by the fact that your 5G connection shows the list ok. If you can specify your model of router, it might be possible to look up online whether there is a relevant setting for it.

There is no way to overcome this in the current versions of Jamulus. There is development work happening to add TCP support for fetching large lists, that would overcome this limitation, but that is a few months away and will not be until Jamulus 4.0

**Reply by Wubbidubbi** [reply]
The workaround mentioned above worked for me.

https://explorer.jamulus.io/

Here you have the server lists. You can look up the ip address and then join directly.

**Reply by Ciccots** [reply]
Thanks for the quick reply, it seems in Italy we have the same issue with different providers, i can see from previous questions that is happening also with "sky" and "vodafone" providers. I have ILIAD.The router seems totally rebranded , mine is called ILIADBOX i don't see any way of understanding brand and type. Also software is proprietary. Anyway, using the server ip works well.I'll wait Jam 4.0

One thing more i have to mention.
Looking inside the router settings i can see there is no way of forwarding ports for UDP protocols since there is a limitation of 16383 ports.
This for sure is an ISP limitation.
thanks

![Iliad](https://github.com/user-attachments/assets/9a6e2393-d2fc-45bf-b49e-75c2a58149f4)

**Reply by QJKX** [reply]
ILIAD are IPv6 - it's probably the IPv4 compatability stuff that doesn't work for this.

This forum seems to be good for Italian Broadband- you might be able to get a proper IPv4 address to fix it:

https://forum.fibra.click/d/38631-richiesta-ipv4-full-stack-per-fibra-iliad/2
